x-pack/plugins/lists/common/schemas/request/create_endpoint_list_item_schema.test.ts

```diff
@@ -0,0 +1,216 @@
+import { CommentsArray } from '../types';
+
+import {
+  CreateEndpointListItemSchema,
+  createEndpointListItemSchema,
+} from './create_endpoint_list_item_schema';
+import { getCreateEndpointListItemSchemaMock } from './create_endpoint_list_item_schema.mock';
+
+describe('create_endpoint_list_item_schema', () => {
+  test('it should validate a typical exception list item request not counting the auto generated uuid', () => {
+    const payload = getCreateEndpointListItemSchemaMock();
+    const decoded = createEndpointListItemSchema.decode(payload);
+    const checked = exactCheck(payload, decoded);
+    const message = pipe(checked, foldLeftRight);
+    delete (message.schema as CreateEndpointListItemSchema).item_id;
+    expect(getPaths(left(message.errors))).toEqual([]);
+    expect(message.schema).toEqual(payload);
+  });
+
+  test('it should not validate an undefined for "description"', () => {
+    const payload = getCreateEndpointListItemSchemaMock();
+    delete payload.description;
+    const decoded = createEndpointListItemSchema.decode(payload);
+    const checked = exactCheck(payload, decoded);
+    const message = pipe(checked, foldLeftRight);
+    expect(getPaths(left(message.errors))).toEqual([
+      'Invalid value "undefined" supplied to "description"',
+    ]);
+    expect(message.schema).toEqual({});
+  });
+
+  test('it should not validate an undefined for "name"', () => {
+    const payload = getCreateEndpointListItemSchemaMock();
+    delete payload.name;
+    const decoded = createEndpointListItemSchema.decode(payload);
+    const checked = exactCheck(payload, decoded);
+    const message = pipe(checked, foldLeftRight);
+    expect(getPaths(left(message.errors))).toEqual([
+      'Invalid value "undefined" supplied to "name"',
+    ]);
+    expect(message.schema).toEqual({});
+  });
+
+  test('it should not validate an undefined for "type"', () => {
+    const payload = getCreateEndpointListItemSchemaMock();
+    delete payload.type;
+    const decoded = createEndpointListItemSchema.decode(payload);
+    const checked = exactCheck(payload, decoded);
+    const message = pipe(checked, foldLeftRight);
+    expect(getPaths(left(message.errors))).toEqual([
+      'Invalid value "undefined" supplied to "type"',
+    ]);
+    expect(message.schema).toEqual({});
+  });
+
+  test('it should not validate a "list_id" since it does not required one', () => {
+    const inputPayload: CreateEndpointListItemSchema & { list_id: string } = {
+      ...getCreateEndpointListItemSchemaMock(),
+      list_id: 'list-123',
+    };
+    const decoded = createEndpointListItemSchema.decode(inputPayload);
+    const checked = exactCheck(inputPayload, decoded);
+    const message = pipe(checked, foldLeftRight);
+    expect(getPaths(left(message.errors))).toEqual(['invalid keys "list_id"']);
+    expect(message.schema).toEqual({});
+  });
+
+  test('it should not validate a "namespace_type" since it does not required one', () => {
```

Comment:
Is this because we're only allowing `agnostic` for endpoint?

Reply:
Yep. It broadens the REST API surface area by adding specific endpoints for this specific list and type but I think this is probably going to be best for us to support it as first class citizen of the system and API at this point and still provide ways for the system to be flexible enough to "pivot" to multiple types of agnostic lists in case that's the direction we move towards in the future.